qiskit_experiments/test/t2hahn_backend.py

```diff
@@ -0,0 +1,317 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+T2HahnBackend class.
+Temporary backend to be used for t2hahn experiment
+"""
+
+import numpy as np
+from numpy import isclose
+from qiskit.providers import BackendV1
+from qiskit.providers.models import QasmBackendConfiguration
+from qiskit.result import Result
+from qiskit_experiments.framework import Options
+from qiskit_experiments.test.utils import FakeJob
+
+# Fix seed for simulations
+SEED = 9000
+
+
+class T2HahnBackend(BackendV1):
+    """
+    A simple and primitive backend, to be run by the T2Hahn tests
+    """
+
+    def __init__(
+        self,
+        p0=None,
+        initialization_error=None,
+        readout0to1=None,
+        readout1to0=None,
+        conversion_factor=1,
+    ):
+        """
+        Initialize the T2Hahn backend
+        """
+        conversion_factor_in_ns = conversion_factor * 1e9 if conversion_factor is not None else None
+        configuration = QasmBackendConfiguration(
+            backend_name="T2Hahn_simulator",
+            backend_version="0",
+            n_qubits=int(1e6),
+            basis_gates=["barrier", "ry", "rx", "delay", "measure"],
+            gates=[],
+            local=True,
+            simulator=True,
+            conditional=False,
+            open_pulse=False,
+            memory=False,
+            max_shots=int(1e6),
+            coupling_map=None,
+            dt=conversion_factor_in_ns,
+        )
+
+        self._t2hahn = p0["T2"]
+        self._a_param = p0["A"]
+        self._frequency = p0["frequency"]
```

Comment:
The frequency is indeed a parameter of the simulation, but it's not part of `p0`, in the sense that it's not something that's estimated by the fitter and is not part of user guesses. So, instead of `p0`, the constructor should have parameters `t2` (or `t2hahn`) and `frequency`.

Reply:
Changed it so we will get as input frequency and T2.